aws/resource_aws_glue_dev_endpoint_test.go

```diff
@@ -0,0 +1,968 @@
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSGlueDevEndpointExists(resourceName, &endpoint),
+
+					resource.TestCheckResourceAttr(resourceName, "worker_type", glue.WorkerTypeG1x),
+				),
+			},
+			{
+				Config: testAccGlueDevEndpointConfig_WorkerType(rName, glue.WorkerTypeG2x),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSGlueDevEndpointExists(resourceName, &endpoint),
+
+					resource.TestCheckResourceAttr(resourceName, "worker_type", glue.WorkerTypeG2x),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccGlueDevEndpoint_disappears(t *testing.T) {
+	var endpoint glue.DevEndpoint
+
+	rName := acctest.RandomWithPrefix(GlueDevEndpointResourcePrefix)
+	resourceName := "aws_glue_dev_endpoint.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSGlueDevEndpointDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccGlueDevEndpointConfig_Basic(rName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSGlueDevEndpointExists(resourceName, &endpoint),
+					testAccCheckResourceDisappears(testAccProvider, resourceAwsGlueDevEndpoint(), resourceName),
+				),
+				ExpectNonEmptyPlan: true,
+			},
+		},
+	})
+}
+
+func testAccCheckAWSGlueDevEndpointExists(resourceName string, endpoint *glue.DevEndpoint) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("not found: %s", resourceName)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("no ID is set")
+		}
+
+		conn := testAccProvider.Meta().(*AWSClient).glueconn
+		output, err := conn.GetDevEndpoint(&glue.GetDevEndpointInput{
+			EndpointName: aws.String(rs.Primary.ID),
+		})
+
+		if err != nil {
+			return err
+		}
+
+		if output == nil {
+			return fmt.Errorf("no Glue Dev Endpoint")
+		}
+
+		*endpoint = *output.DevEndpoint
+
+		return nil
+	}
+}
+
+func testAccCheckAWSGlueDevEndpointDestroy(s *terraform.State) error {
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_glue_dev_endpoint" {
+			continue
+		}
+
+		conn := testAccProvider.Meta().(*AWSClient).glueconn
+		output, err := conn.GetDevEndpoint(&glue.GetDevEndpointInput{
+			EndpointName: aws.String(rs.Primary.ID),
+		})
+
+		if err != nil {
+			if isAWSErr(err, glue.ErrCodeEntityNotFoundException, "") {
+				return nil
+			}
+			return err
+		}
+
+		endpoint := output.DevEndpoint
+		if endpoint != nil && aws.StringValue(endpoint.EndpointName) == rs.Primary.ID {
+			return fmt.Errorf("the Glue Dev Endpoint %s still exists", rs.Primary.ID)
+		}
+
+		return nil
+	}
+
+	return nil
+}
+
+func testAccGlueDevEndpointConfig_Base(rName string) string {
+	return fmt.Sprintf(`
+resource "aws_iam_role" "test" {
+  name               = "AWSGlueServiceRole-%s"
```

Comment:
lets use `%[1]s` and `%[1]q` for the one below and use `rName` once in the fmt.Sprintf params